### PR TITLE
ashahba/Start to upgrade r0.19 branch to TF v1.15.0

### PR DESCRIPTION
**Pull request by ashahba**

### CODE REVIEW

**Comment by sayantan-nervana** [comment]
@googlebot I consent.

File: build_ngtf.py [hunk 1]
```diff
@@ -38,12 +38,12 @@ def version_check(use_prebuilt_tensorflow):
...
     bazel_ver = get_bazel_version()
     got_correct_bazel_version = False
-    if (int(bazel_ver[1]) >= 24 and int(bazel_ver[1]) <= 25):
+    if (int(bazel_ver[1]) > 24 and int(bazel_ver[1]) <= 25):
         if (int(bazel_ver[2]) >= 1 and int(bazel_ver[2]) <= 2):
             got_correct_bazel_version = True
 
     if not got_correct_bazel_version:
-        raise Exception("Need bazel 0.24.1 < version < 0.25.2 \n" + "Got: " +
+        raise Exception("Need bazel 0.24.x < version <= 0.25.2 \n" + "Got: " +
```
Comment (kanvi-nervana) [inline build_ngtf.py]: Should it be 0.24.1 <= version <= 0.25.2

File: test/python/tensorflow/python_tests_list.txt [hunk 2]
```diff
@@ -497,7 +497,9 @@ softmax_op_test.SoftmaxTest.testHalf
 #softmax_op_test.SoftmaxTest.testHalfGPU
 softmax_op_test.SoftmaxTest.testLargeDims
 
-spacetodepth_op_test.SpaceToDepthTest.testAgainstTranspose
+# Test failes due to DT_INT8 not implemented during TF v1.15.0 upgrade
```
Comment (kanvi-nervana) [inline test/python/tensorflow/python_tests_list.txt]:
```suggestion
# Test fails due to DT_INT8 not being implemented for TF v1.15.0
```

File: build_ngtf.py (hunk 1 above)
Comment (ashahba) [reply build_ngtf.py]: But we want `0.24.1 < version <= 0.25.2`

Comment (kanvi-nervana) [reply build_ngtf.py]: The check in 42 seems to be checking for =1 as well

**Comment by sayantan-nervana** [comment]
Ignoring the binary and mac pipeline.
binary pipeline will not pass on 168 anyway as it is using gcc4.8 and tf1.15 needs higher gcc

File: test/ci/buildkite/ngtf-cpu_ubuntu-external-fork.yaml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    agents:
+    - "queue=cpu-external"
+    parallelism: 1
+
+  - wait
+
+  - command: |
+      source /localdisk/buildkite/artifacts/$BUILDKITE_BUILD_ID/venv/bin/activate
+      pip install -U \
+        /localdisk/buildkite/artifacts/$BUILDKITE_BUILD_ID/tensorflow/tensorflow-1.14.0-cp36-cp36m-linux_x86_64.whl
```
Comment (kanvi-nervana) [inline test/ci/buildkite/ngtf-cpu_ubuntu-external-fork.yaml]: Should we add the pip install pip==19.3.1 in this yaml files as well ?

Comment (kanvi-nervana) [reply test/ci/buildkite/ngtf-cpu_ubuntu-external-fork.yaml]: NVM I saw it is added

File: .gitignore [hunk 4]
```diff
@@ -12,3 +12,5 @@ cpu_codegen/
 
 .*.swp
 .nfs*
+build_cmake
+test/**/dump.sh
```
Comment (kanvi-nervana) [inline .gitignore]: What file is this?

Comment (ashahba) [reply .gitignore]: The file gets generated if you run `model` tester manually so we don't want to accidentally add it to the push.

**Pull request merged**